src/kv/apply_changes.h

```diff
@@ -70,7 +71,15 @@ namespace kv
     }
 
     bool ok = true;
-    if (has_writes)
+
+    if (expected_rollback_count.has_value())
+    {
+      // Assumes that all maps in the transaction point to the same store
+      auto store = changes.begin()->second.map->get_store();
+      ok = store->check_rollback_count(expected_rollback_count.value());
```

Comment:
Think this deserves a comment, explaining the secret trick here is that we're checking this _while holding locks on some maps_. So any rollback which tries to happen _after_ this check will get queued (waiting for _this_ block to release the map locks that the rollback requires), and thus we get mutual exclusion.

Reply:
Good point, I will add a comment to that effect.